services/src/main/java/com/broadleafcommerce/subscriptionoperation/domain/ExistingSubscriptionItemDetails.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2024 Broadleaf Commerce
+ *
+ * Licensed under the Broadleaf End User License Agreement (EULA), Version 1.1 (the
+ * "Commercial License" located at http://license.broadleafcommerce.org/commercial_license-1.1.txt).
+ *
+ * Alternatively, the Commercial License may be replaced with a mutually agreed upon license (the
+ * "Custom License") between you and Broadleaf Commerce. You may not use this file except in
+ * compliance with the applicable license.
+ *
+ * NOTICE: All information contained herein is, and remains the property of Broadleaf Commerce, LLC
+ * The intellectual and technical concepts contained herein are proprietary to Broadleaf Commerce,
+ * LLC and may be covered by U.S. and Foreign Patents, patents in process, and are protected by
+ * trade secret or copyright law. Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained from Broadleaf Commerce, LLC.
+ */
+package com.broadleafcommerce.subscriptionoperation.domain;
+
+import com.broadleafcommerce.cart.client.domain.CartItem;
+
+import java.io.Serial;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+import lombok.Data;
+
+/**
+ * Contains details describing an existing {@link SubscriptionItem} when entering a
+ * modify-subscription flow such as for edit, upgrade, or downgrade. This is stored on the created
+ * {@link CartItem} for use by downstream systems that handle fulfillment like Order Operations
+ * Service in comparing the subscription before and after the modifications.
+ */
+@Data
+public class ExistingSubscriptionItemDetails implements Serializable {
```

Comment:
```suggestion
public class ExistingSubscriptionItemDetail implements Serializable {
```
nitpick: the domain itself represent one single subscription item detail so it should be singular